webrecorder/browsertrix
# Invite user to an archive

Invites users to access an archive

https://github.com/ikreymer/browsertrix-cloud/issues/31
Possibly blocked by https://github.com/ikreymer/browsertrix-cloud/issues/35

### Manual testing
1. Run app with `yarn start-dev`
2. Go to your archives http://localhost:9870/archives and choose an archive
3. Click "Members"
4. Click "Add Member"
5. Enter email you can access and click "Invite"
6. Copy invite token from email and go to `http://localhost:9870/join/{invite token}?email={invited email}`
7. Enter password and sign up
8. Accept invitation. Verify you are redirected to archives page

### Screenshots
Archive members tab:
<img width="931" alt="Screen Shot 2021-12-02 at 11 21 12 AM" src="https://user-images.githubusercontent.com/4672952/144488752-32cfc6cd-29fa-492d-a27c-d40b014c0d48.png">
the grayed out "Member"s are emails invited before adding the `name` field

Add new member form:
<img width="923" alt="Screen Shot 2021-11-30 at 4 44 11 PM" src="https://user-images.githubusercontent.com/4672952/144151184-55a8e8bc-5d06-4643-a783-0360ce92bb08.png">

Join invite page for new users:
<img width="456" alt="Screen Shot 2021-12-02 at 11 03 57 AM" src="https://user-images.githubusercontent.com/4672952/144488662-d864ba35-a80a-45a3-851f-9b5c719ef552.png">

Accept invite view with invalid invite:
<img width="440" alt="Screen Shot 2021-11-30 at 4 39 51 PM" src="https://user-images.githubusercontent.com/4672952/144150811-8e8cccc1-6891-4bed-8e83-b1a7fd7b733b.png">
